API.Test/ReviewServiceTest.cs

```diff
@@ -0,0 +1,279 @@
+using Reviews.API.Services;
+using Reviews.Domain.Entities.Factories;
+using Reviews.Domain.Entities;
+using Reviews.Infrastructure.Kafka;
+using Moq;
+using Reviews.Domain.Exceptions;
+using Reviews.API.DTOs;
+using Reviews.Infrastructure.Repositories;
+
+namespace API.Test
+{
+    public class ReviewServiceTest
+    {
+        [Fact]
+        public async void CreateReview_RestaurantReview_ValidRequest_ShouldResolveCorrectReviewFactoryAndCreateReview()
+        {
+            // Arrange
+            var mockKafkaProducer = new Mock<IKafkaProducer>();
+            var mockRepo = new Mock<IReviewRepository>();
+
+            var mockFactoryResolver = new Mock<Func<string, IReviewFactory>>();
+            mockFactoryResolver
+                .Setup(resolver => resolver("restaurant"))
+                .Returns(new RestaurantReviewFactory());
+
+            var reviewService = new ReviewService(mockKafkaProducer.Object, mockFactoryResolver.Object, mockRepo.Object);
+
+            var type = "restaurant";
+
```

Comment:
Og det samme her som tidligere kommentar, hvis det kan lægges uden for test metoderne så gør det :)